hibernate-core/src/main/antlr/hql.g

```diff
@@ -846,6 +846,13 @@ path
 	: identifier ( DOT^ { weakKeywords(); } identifier )*
 	;
 
+treatAsPath
+	: path
+	| i:IDENT! OPEN! p:treatAsPath AS! a:path! CLOSE! ( DOT^ path )? {i.getText().equalsIgnoreCase("treat") }? {
```

Comment:
Oh, nevermind about the IDENT + semantic predicate.  We did that to avoid making TREAT a keyword.